packages/rOpenScPCA/R/calculate-clusters.R

```diff
@@ -110,42 +141,34 @@ calculate_clusters <- function(
 }
 
 
-
 #' Extract a principal components (PC) matrix from either a SingleCellExperiment
 #' or a Seurat object.
 #'
-#' This function first determines if the provided object is a SingleCellExperiment or
-#' Seurat object, and then extract the PC matrix. If no name for the PC matrix is provided,
-#' this function will assume the name of "PCA" for SingleCellExperiment objects, and
-#' "pca" for Seurat objects.
+#' This function extract the PC matrix from either a SingleCellExperiment or Seurat object, as
+#' determined by the object_type argument.
 #'
 #' @import SingleCellExperiment
 #' @import methods
 #'
 #' @param sc_object Either a SingleCellExperiment or Seurat object
-#' @param pc_name Optionally, the name of the PC matrix in the object. If this is
-#' not provided, the name "PCA" is assumed for SingleCellExperiment objects, and
-#' "pca" for Seurat objects.
+#' @param object_type Whether the object is a SingleCellExperiment ("sce") or a Seurat ("seurat") object
```

Comment:
I don't like this argument; you should check that the object is what was expected anyway, so I would leave the simpler form and check the actual object as you did before.